content/usgs-packages/Application_Part3.Rmd

```diff
@@ -0,0 +1,182 @@
+
+```{r filter-data, addToggle="ON", message=FALSE, warning=FALSE}
+library(dplyr)
+
+nitrogen_site1 <- filter(nitrogen_data, site_no == sites[1])
+phosphorus_site1 <- filter(phosphorus_data, site_no == sites[1])
+
+huc_site1 <- filter(sb_sites_info, site_no == sites[1])$huc_cd # corresponding HUC8
+precip_site1 <- filter(precip_data_long, huc8 == huc_site1)
+```
+
+## Create time series plots
+
+Create a separate time series plot of precipitation, nitrogen, and phosphorus for each site using the separated data frames. See `?layout` to see how to include all three in one graphics device. 
+
+```{r time-series-plots, addToggle="ON"}
+layout(matrix(1:3, nrow=3))
+plot(precip_site1$DateTime, precip_site1$precip,
+     col="red", pch=20, xlab = "Time", ylab = "Precip accumulation, in")
+plot(nitrogen_site1$sample_dt, nitrogen_site1$result_va, 
+     col="green", pch=20, xlab = "Time", ylab = "Nitrogren concentration, mg/l")
+plot(phosphorus_site1$sample_dt, phosphorus_site1$result_va,
+     col="blue", pch=20, xlab = "Time", ylab = "Phosphorus concentration, mg/l")
+```
+
+## Automate the plots for all sites
+
+Now that we have established the code required to make all three timeseries plots for each site, automate this so it happens for each site (a.k.a. make a [for loop](/intro-curriculum/reproduce/#looping)). 
+
+```{r automate-time-series-plots, addToggle="ON"}
+for(i in sites){
+  huc_site_i <- filter(sb_sites_info, site_no == i)$huc_cd # corresponding HUC8
+
+  precip_site_i <- filter(precip_data_long, huc8 == huc_site1)
+  nitrogen_site_i <- filter(nitrogen_data, site_no == i)
+  phosphorus_site_i <- filter(phosphorus_data, site_no == i)
+  
+  layout(matrix(1:3, nrow=3))
+  plot(precip_site_i$DateTime, precip_site_i$precip,
+       col="red", pch=20, xlab = "Time", ylab = "Precip accumulation, in",
+       main = paste("Site", i))
+  plot(nitrogen_site_i$sample_dt, nitrogen_site_i$result_va, 
+       col="green", pch=20, xlab = "Time", ylab = "Nitrogren concentration, mg/l")
+  plot(phosphorus_site_i$sample_dt, phosphorus_site_i$result_va,
+       col="blue", pch=20, xlab = "Time", ylab = "Phosphorus concentration, mg/l")
+}
+```
+
+## Map cumulative precip
+
+Finally, create a map that summarizes all HUC yearly cumulative precipitation at once. This will take some data massaging first: determine the cumulative annual (by water year) precipitation for each HUC.
+
+```{r cumulative-precip, addToggle="ON"}
+precip_annual <- precip_data_long %>%
+  rename(dateTime = DateTime) %>% # need to rename to use with addWaterYear
+  addWaterYear() %>% 
+  group_by(waterYear, huc8) %>% 
+  summarize(cumulative_precip = sum(precip))
+```
+
+Now, map the HUCs and color based on precipication. 
+
+HMMMMMMMMMM I HAVEN'T DECIDED THE CLEANEST WAY TO DO THIS YET.
```

Comment:
FYI @aappling-usgs I'm trying to work this out now. So just skip for now :) I can merge with a different PR if need-be.